src/users_db.rs

```diff
@@ -151,3 +151,86 @@ impl UsersDb {
         self.connection.execute("DELETE FROM users WHERE id=$1", &[&id])
     }
 }
+
+describe! user_builder_tests {
+
+    it "should build a user correctly" {
+        use super::crypto::digest::Digest;
+        use super::crypto::md5::Md5;
+
+        let user = UserBuilder::new()
+            .id(1)
+            .name("Mr Fox")
+            .email("fox@mozilla.org")
+            .password("pass12345678")
+            .finalize()
+            .unwrap();
+
+        assert_eq!(user.id, Some(1));
+        assert_eq!(user.name, "Mr Fox");
+        assert_eq!(user.email, "fox@mozilla.org");
+        let mut md5 = Md5::new();
+        md5.input_str("pass12345678");
+        assert_eq!(user.password, md5.result_str());
+    }
+
+    failing "should panic if invalid user" {
```

Comment:
We would need to test the different failure reasons: empty or short password, empty name, empty email, malformed email. But that can be done in a follow up.